rohmu/object_storage/s3.py

```diff
@@ -137,6 +137,7 @@ def __init__(
             if proxy_info:
                 proxy_url = get_proxy_url(proxy_info)
                 custom_config["proxies"] = {"https": proxy_url}
+            custom_config["use_dualstack_endpoint"] = True
```

Comment:
Should this be provided as a parameter so that calling code is able to control the value in case this ends up causing some problems?

Reply:
I added a configuration option to turn this off, but I'd personally prefer keeping it enabled by default: IPv6 is nowdays well supported, if available and configured.

Reply:
That actually reminds me, was this validated against non-AWS implementations of S3? I at least have no idea what's the state of IPv6 with those.

Reply:
The config only applies to default botocore addresses, and is not used if one passes explicit host/port information.